File: clients/rust/model/src/mine.rs
use crate::*;
#[derive(Clone, Debug, trans::Trans)]
pub struct Mine {
    pub player_id: i32,
    pub position: Vec2F64,
    pub size: Vec2F64,
    pub state: MineState,
    pub timer: Option<f64>,
    pub trigger_radius: f64,
    pub explosion_params: ExplosionParams,
}

File: clients/haskell/src/RAIC/Model/Mine.hs
module RAIC.Model.Mine where

import           GHC.Generics               (Generic)
import           RAIC.Model.ExplosionParams (ExplosionParams)
import           RAIC.Model.MineState
import           RAIC.Model.Vec2Double      (Vec2Double)
import           RAIC.StreamWrapper         (Trans)

data Mine = Mine {
  player_id        :: Int,
  position         :: Vec2Double,
  size             :: Vec2Double,
  state            :: MineState,
  timer            :: Maybe Double,
  trigger_radius   :: Double,
  explosion_params :: ExplosionParams
} deriving (Generic, Show)

instance Trans Mine
